src/test/scala/mesosphere/marathon/api/TaskKillerTest.scala

```diff
@@ -84,6 +86,48 @@ class TaskKillerTest extends MarathonSpec
     toKillCaptor.getValue shouldEqual Map(appId -> tasksToKill)
   }
 
+  test("KillRequested with killCount") {
+    val f = new Fixture
+    val appId = PathId(List("app"))
+    val task1 = MarathonTestHelper.runningTaskForApp(appId)
+    val task2 = MarathonTestHelper.runningTaskForApp(appId)
+    val tasksToKill = List(task1, task2)
+
+    when(f.tracker.hasAppTasksSync(appId)).thenReturn(true)
+    when(f.groupManager.group(appId.parent)).thenReturn(Future.successful(Some(Group.emptyWithId(appId.parent))))
+    when(f.groupManager.app(appId)).thenReturn(Future.successful(Some(AppDefinition(appId))))
+
+    val expectedDeploymentPlan = DeploymentPlan.empty
+    val toKillCaptor = ArgumentCaptor.forClass(classOf[Map[PathId, Iterable[Task]]])
+    when(f.groupManager.update(
+      any[PathId],
+      any(classOf[Group => Group]),
+      any[Timestamp],
+      any[Boolean],
+      toKillCaptor.capture()
+    )).thenReturn(Future.successful(expectedDeploymentPlan))
+
+    val result = f.taskKiller.killAndScale(appId, (tasks) => tasksToKill, force = true, killCount = 1)
+    result.futureValue shouldEqual expectedDeploymentPlan
+    toKillCaptor.getValue.size shouldEqual 1
+    tasksToKill contains toKillCaptor.getValue.values.take(1)
+  }
+
+  class IterableEquals[A](var original: Iterable[A]) extends ArgumentMatcher[Iterable[A]] {
```

Comment:
This is necessary since the TaskKiller test internally passes around generic `Iterable[Task]` objects. So it's no longer possible to just use equality operators.